transform/set_slice_mangler_test.go

```diff
@@ -0,0 +1,97 @@
+package transform
+
+import (
+	"reflect"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/vimeo/dials/ptrify"
+)
+
+func TestSetSliceManglerMangle(t *testing.T) {
+	m := SetSliceMangler{}
+	sf := reflect.StructField{
+		Type: reflect.TypeOf(map[string]struct{}{}),
+	}
+	sfs, err := m.Mangle(sf)
+
+	require.NoError(t, err)
+
+	assert.Equal(t, reflect.TypeOf([]string{}), sfs[0].Type)
+}
+
+func TestSetSliceManglerUnmangle(t *testing.T) {
+	cases := map[string]struct {
+		StructFieldType reflect.Type
+		SrcValue        interface{}
+		AssertFunc      func(testing.TB, interface{})
+		ExpectedErr     string
```

Comment:
We can add an `expectedMap` field with the values we are expecting and assert `reflect.DeepEqual(expectedMap, set)` instead so we only need one assert statement in the test body instead of the assert function for each case

Reply:
I like that a lot!  Done.